app/services/optout/OptOutService.scala

```diff
@@ -157,43 +135,54 @@ class OptOutService @Inject()(itsaStatusUpdateConnector: ITSAStatusUpdateConnect
 
     def findAnyFailOrFirstSuccess(responses: Seq[OptOutUpdateResponse]): OptOutUpdateResponse = {
       responses.find {
-          case _: OptOutUpdateResponseFailure => true
-          case _ => false
+        case _: OptOutUpdateResponseFailure => true
+        case _ => false
       }.getOrElse(responses.head)
     }
 
     val result = for {
       intentTaxYear <- OptionT(fetchSavedIntent())
       yearsToUpdate <- OptionT(Future.successful(Option(optOutProposition.optOutYearsToUpdate(intentTaxYear))))
-      responsesSeqOfFutures  <- OptionT(Future.successful(Option(makeUpdateCalls(yearsToUpdate))))
+      responsesSeqOfFutures <- OptionT(Future.successful(Option(makeUpdateCalls(yearsToUpdate))))
       responsesSeq <- OptionT(Future.sequence(responsesSeqOfFutures).map(v => Option(v)))
       finalResponse <- OptionT(Future.successful(Option(findAnyFailOrFirstSuccess(responsesSeq))))
     } yield finalResponse
 
     result.getOrElse(OptOutUpdateResponseFailure.defaultFailure())
   }
 
+  def nextUpdatesPageOptOutViewModel(proposition: OptOutProposition)(implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext): Option[OptOutViewModel] = {
+    proposition.optOutPropositionType.flatMap {
+      case p: OneYearOptOutProposition => Some(OptOutOneYearViewModel(oneYearOptOutTaxYear = p.intent.taxYear, state = p.state()))
+      case _: MultiYearOptOutProposition => Some(OptOutMultiYearViewModel())
+    }
+  }
+
   def nextUpdatesPageOptOutViewModel()(implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext): Future[Option[OptOutViewModel]] = {
+    fetchOptOutProposition().map(nextUpdatesPageOptOutViewModel)
+  }
+
+  def nextUpdatesPageOptOutWithChecks()(implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext)
+  : Future[(Option[OptOutViewModel], NextUpdatesQuarterlyReportingContentChecks)] = {
+
     fetchOptOutProposition().map { proposition =>
-      proposition.optOutPropositionType.flatMap {
-        case p: OneYearOptOutProposition => Some(OptOutOneYearViewModel(oneYearOptOutTaxYear = p.intent.taxYear, state = p.state()))
-        case _: MultiYearOptOutProposition => Some(OptOutMultiYearViewModel())
-      }
+      (nextUpdatesPageOptOutViewModel(proposition),
```

Comment:
not covered by unit test
![Screenshot 2024-07-04 at 11 42 31](https://github.com/hmrc/income-tax-view-change-frontend/assets/165788042/aaf03e7f-ff7a-4a6a-907a-0605e7cf1b5d)